config/confighttp/confighttp.go

```diff
@@ -203,6 +203,9 @@ type HTTPServerSettings struct {
 	// Endpoint configures the listening address for the server.
 	Endpoint string `mapstructure:"endpoint"`
```

Comment:
If needed to extend with transport, we should "embed" (using mapstructure `squash`) https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confignet/confignet.go#L22 into this config.